core/src/Cuda/Kokkos_Cuda_Team.hpp

```diff
@@ -667,15 +723,41 @@ parallel_reduce(Impl::ThreadVectorRangeBoundariesStruct<
   (void)loop_boundaries;
   (void)closure;
   (void)result;
-  KOKKOS_IF_ON_DEVICE(
-      (result = ValueType();
 
-       for (iType i = loop_boundaries.start + threadIdx.x;
-            i < loop_boundaries.end; i += blockDim.x) { closure(i, result); }
-
-       Impl::CudaTeamMember::vector_reduce(Kokkos::Sum<ValueType>(result));
-
-       ))
+  KOKKOS_IF_ON_DEVICE((
+      using functor_analysis_type = typename Impl::FunctorAnalysis<
+          Impl::FunctorPatternInterface::REDUCE,
+          TeamPolicy<typename Impl::CudaTeamMember::execution_space>, Closure,
+          ValueType>;
+
+      constexpr bool is_reducer_closure =
+          functor_analysis_type::has_join_member_function &&
+          functor_analysis_type::has_init_member_function;
+
+      using ReducerSelector =
+          typename Kokkos::Impl::if_c<is_reducer_closure, Closure,
+                                      Sum<ValueType>>::type;
```

Comment:
See above.

Reply:
Removed and replaced by `FunctorAnalysis::Reducer` usage.